File: 剑指Offer/43.1-n整数中n出现的次数.cpp
class Solution {
public:
    int countDigitOne(int n) {
        int counts = 0;
        for (long long i = 1; i <= n; i *= 10) {
            long long divider = i * 10;
            counts += (n / divider) * i + min(max(n % divider - i + 1, 0LL), i);
        }
        return  counts;
    }
};

File: 剑指Offer/43.1-n整数中n出现的次数.py
class Solution:
    def countDigitOne(self, n: int) -> int:
        counts = 0
        i = 1
        while i <= n:
            divider = i * 10
            counts += (n // divider) * i + min(max(n % divider - 10 + 1, 0), i)
            i *= 10
        return counts